src/basic/mountpoint-util.c

```diff
@@ -187,7 +187,7 @@ int fd_is_mount_point(int fd, const char *filename, int flags) {
 
         /* Insist that the specified filename is actually a filename, and not a path, i.e. some inode further
          * up or down the tree then immediately below the specified directory fd. */
-        if (!filename_possibly_with_slash_suffix(filename))
+        if (!FLAGS_SET(flags, AT_EMPTY_PATH) && !filename_possibly_with_slash_suffix(filename))
```

Comment:
see: https://github.com/systemd/systemd/pull/24179

unfortunately I see no way how to make fd_is_mount_point() work on non-dir inodes on old kernels…